### ISSUE TITLE
Rule Change: Check `yield await`

### ISSUE THREAD

**Issue by fisker** [eslint/eslint#15447]
### What rule do you want to change?

no-return-await

### What change to do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
async function * foo () {
    yield await promise;
}


`await` in this case is redundant.
```

### What does the rule currently do for this code?

Nothing

### What will the rule do after it's changed?

Report a problem.

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

The existing `no-return-await` rule name may not suitable for this, a new rule? rename?

**Comment by nzakas** [comment]
You didn’t explain why we would want to make such a change. What’s the reasoning?
